pkg/segment/structs/evaluationstructs.go

```diff
@@ -1533,6 +1533,52 @@ func (self *ValueExpr) EvaluateValueExprAsString(fieldToValue map[string]utils.C
 	return str, nil
 }
 
+func handleCaseFunction(self *ConditionExpr, fieldToValue map[string]utils.CValueEnclosure) (string, error) {
+
+	for _, cvPair := range self.ConditionValuePairs {
+		res, err := cvPair.Condition.Evaluate(fieldToValue)
+		if err != nil {
+			continue
+		}
```

Comment:
Do you `continue` on line 1541 because `Evaluate()` may have failed because a required field didn't have a value? I think we'll want to detect that separately, since we do want to `continue` in that case, but other times we might get a different kind of error during `Evaluate()` and in that case we would want to log and return the error